### PR TITLE
[Cute,Fwd] improved block sparsity

### PR DESCRIPTION
**Pull request by reubenconducts**
This PR introduces a number of quality-of-life changes regarding block sparsity computation, including enabling `tvm-ffi` for the block sparsity computation interface. 

We expand the signature of `mask_mods` to 
```python
def generic_mask_mod(
	batch, head, m_idx, n_idx, seqlen_info, aux_tensors
)
```
to match those of `score_mods`, in particular enabling offset `mask_mods`.

### CODE REVIEW

**Comment by reubenconducts** [comment]
@drisspg @tridao minor QoL improvements surrounding compute block sparsity, plus change of mask mod signature to mimic score mod

**Review by drisspg — COMMENTED** [review]
LGTM

**Comment by reubenconducts** [comment]
@tridao should be ready to go

File: flash_attn/cute/flash_fwd.py [hunk 1]
```diff
@@ -2004,7 +2004,7 @@ def mma(
                     else FastDivmodDivisor(seqlen.seqlen_k),
                 )
 
-            mask = AttentionMaskCls(seqlen.seqlen_q, seqlen.seqlen_k)
+            mask = AttentionMaskCls(seqlen)
```
Comment (tridao) [inline flash_attn/cute/flash_fwd.py:2007]: I think we'll need to change the bwd kernels to construct the attention mask this way too

**Comment by tridao** [comment]
LGTM except we'll need to change bwd when constructing AttentionMaskCls. After that we can merge.

**Comment by tridao** [comment]
Oh there's one linter error as well
```
--- a/flash_attn/cute/compute_block_sparsity.py
+++ b/flash_attn/cute/compute_block_sparsity.py
@@ -4,7 +4,7 @@ from typing import Callable, Optional, Tuple
 import cutlass
 import cutlass.cute as cute
 import torch
-from cutlass import Boolean, Int8, Int32, Uint32, const_expr
+from cutlass import Boolean, Int8, Int32, const_expr
```
